app/models/events/review.rb

```diff
@@ -56,5 +56,15 @@ def build_review_turnaround
 
       Builders::ReviewTurnaround.call(review_request)
     end
+
+    def build_second_review_turnaround
+      return unless second_review_with_different_users?
+
+      Builders::SecondReviewTurnaround.call(self)
+    end
+
+    def second_review_with_different_users?
+      pull_request.reviews.select(:owner_id).distinct.count == 2
```

Comment:
@santiagovidal @santib @SandroDamilano guys, i am not sure about this query but
`Events::Review.joins(:pull_request).where(pull_request_id: self.pull_request_id).select(:owner_id).distinct.count == 2` this one i think it is not right because emulation of a class method right?

Reply:
I may be missing something but I don't see anything wrong with any of both 🤔 

I do want to point out that I think we need another condition, because a `SecondReviewTurnaround` will be created every time a new review is submitted by any of the two original reviewers.
Let's say someone creates a PR and I review it first. I give it my approve.
Then you review it but request some changes. A `SecondReviewTurnaround` is created.
The comments are addressed and you review it again, approving it. Another `SecondReviewTurnaround` will be created, because there have been still 2 reviewers so far, you and me.
We probably need to add a condition that checks that the reviewer has not made any other review to this PR. That should be enough.